ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/DynamicPivotBlockManager.java

```diff
@@ -93,9 +93,9 @@ public void switchToNewPivotBlock(final BiConsumer<BlockHeader, Boolean> onSwitc
           LOG.info(
               "Select new pivot block {} {}", blockHeader.getNumber(), blockHeader.getStateRoot());
           syncState.setCurrentHeader(blockHeader);
+          lastPivotBlockFound = Optional.empty();
```

Comment:
Since this is getting set in both `check` and `switchToNewPivotBlock` , perhaps we should use a `final AtomicReference<BlockHeader> instead of a mutable `Optional<BlockHeader>`

IDK if the snap pipeline might be calling these concurrently, but it seems possible.